src/server/compiler/indexer.cpp

```
@@ -1594,6 +1666,25 @@ kota::task<> Indexer::run_index_task(std::uint32_t server_path_id,
+            if(it == workspace.shards.end() || !it->second.matches_content(session->text)) {
+                on_session_unservable(server_path_id);
+            }
+        }
+    }
+    // Wake await_attempt waiters after the unservable escalation above, so
+    // they re-derive their route against the settled state. A requeue kept
+    // the entry alive under a fresh ticket — the waiters stay parked for
+    // that newer attempt.
+    if(!reindex_reasons.contains(server_path_id)) {
+        settle_attempt_waits(server_path_id);
+    }
```

Cppcheck (2.21.0)

[error] Found an exit path from function with non-void return type that has missing return statement

(missingReturn)